src/Operation/MapReduce.php

```diff
@@ -264,7 +264,7 @@ public function execute(Server $server)
 
         $cursor = $hasOutputCollection
             ? $server->executeReadWriteCommand($this->databaseName, $command, $options)
-            : $server->executeReadCommand($this->databaseName, $command, $options);
+            : $server->executeCommand($this->databaseName, $command, $options);
```

Comment:
Please correct me if I'm wrong, but [Server::executeCommand](https://github.com/mongodb/mongo-php-driver/blob/e65d114b405ea0fd3f85eab6ef113d782d90c1fc/src/MongoDB/Server.c#L50) uses `PHONGO_COMMAND_RAW` (`0x07`). The way I read the `readConcern` check in [`phongo_execute_command`](https://github.com/mongodb/mongo-php-driver/blob/e65d114b405ea0fd3f85eab6ef113d782d90c1fc/php_phongo.c#L949), it will parse the read concern when using `PHONGO_COMMAND_RAW`.

However, when looking at `mongoc-client-c`, [it only applies a `readConcern` option for `MONGOC_CMD_READ` and `MONGOC_CMD_RW`](https://github.com/mongodb/mongo-c-driver/blob/b37f263e2286ad20c656df1e96fa106377fd7c7a/src/libmongoc/src/mongoc/mongoc-client.c#L1982).

Looking at the `enum` we declare in [`phongo.h`](https://github.com/mongodb/mongo-php-driver/blob/e65d114b405ea0fd3f85eab6ef113d782d90c1fc/php_phongo.h#L120), I believe the declaration for `PHONGO_COMMAND_RAW` should change from `0x07` to `0x06` to not support the `readConcern` option, just the way libmongoc does.

In summary, your point is absolutely correct, as libmongoc does not parse a `readConcern` option when running the command through `executeCommand` vs. `executeReadCommand`. Given the history of `mapReduce`, I'd also be surprised if it started supporting a `readConcern` option. If that were the case however, there would be more changes necessary in both libmongoc and the driver here, so I think we should be safe ignoring this for now. Regardless of the above, I've added a comment to explain why we use `executeCommand` instead of `executeReadCommand`.